### ISSUE TITLE
Usability issue with Save Image saves a WebP image, unable to upload back to WordPress

### ISSUE THREAD

**Issue by joehoyle** [humanmade/altis-media#69]
We recently had someone report a problem with upload a JPG file to Altis. They get the message "Sorry, this file type is not permitted for security reasons.". Turns out, this is a `.jpg` that is actually a webp image.

Because we serve WebP images to `.jpg` files, if anyone saves an image locally they will _think_ they have a JPG, but they actually have a WebP. If they then try upload that image to WordPress, it will fail with the above error.

I don't know if there's much we can do to fix this, but maybe we could atleast improve that error message so it's more clear what the problem is.

**Comment by joehoyle** [comment]
Bumping to v5, as I dont think this is realistic for v4

**Comment by roborourke** [comment]
Do we not just need to filter WP to accept the `image/webp` mime type? Or will tachy not handle it properly either

**Comment by joehoyle** [comment]
Yeah, I don't think we want to allow uploading webp, or rather I don't think that will make anything better. Right now Tachyon would not touch such images (actually, it might because this issue is also about webp images _actually_ having a .jpg extension), so WordPress will serve the .webp image as usual, which will only working in supporting browsers.

Ideally, I don't think we'd accept .webp uploads, like, say, `.tiff` which again I think will only work in some browsers, and have shaky support in the image editing UI etc.

**Comment by roborourke** [comment]
Right I see, may be image magick supports converting it then? Could be another avenue of exploration.

**Comment by goldenapples** [comment]
Is there any chance we could add a query parameter that forces bypassing the webp alternate file (or is there one in place already)?

I'd be happy with adding something like `?raw` onto the image url to download the source image if necessary. Right now pulling down production images means running them through a conversion step before uploading them into my local and I'd love a simpler way of doing that.

**Comment by roborourke** [comment]
@goldenapples if you replace the URLs with the S3 uploads URL you wouldn’t need to do any post processing. What’s your workflow for localising the images?

**Comment by joehoyle** [comment]
If we added `?raw` onto the source image, we woudnb't be displaying a webp file which is what we _want_ to do. In this case, the client was just grabbing the image from the front-end of the site I think, which by definition wants to be the webp file. I don't think there's a way to swap out the saved image when someone does "Save Image As" in the browser.

**Comment by rmccue** [comment]
We might be able to set the `Content-Disposition` header to rename the saved filename. e.g. if we had `Content-Disposition: inline; filename="some-image.jpeg.webp"` it would at least fail the filename extension checks earlier.

**Comment by goldenapples** [comment]
> If we added ?raw onto the source image, we woudnb't be displaying a webp file which is what we want to do. 

well, yeah, sorry I wasn't clear... I didn't mean serving the images as jpg on the site, just providing an easy way to download them as jpgs. 

In this case Rob's suggestion of replacing /tachyon/ with /uploads/ in the URL to get the original file works perfectly... not sure why I hadn't thought of that before. 

> What’s your workflow for localising the images?

I'm just grabbing images by right-clicking a webpage and "saving as" when I need dummy content to test media functionality locally - this is not a critical feature to me at all, just a minor annoyance, and replacing the URL works fine for me.

**Comment by roborourke** [comment]
Cheers Than, I was going to suggest downloading the media backups from Vantage too but if you just need a few images ad-hoc then all good, we'll just focus on a solution like what @rmccue suggested with setting the mime type on download.
